proof/crefine/AARCH64/ADT_C.thy

```diff
@@ -140,6 +140,12 @@ where
     then None
     else Some (ptr_val vcpuptr, to_bool vcpuactive)"
 
+definition ccur_fpu_to_H :: "tcb_C ptr \<Rightarrow> machine_word option" where
```

Comment:
this is really any TCB pointer to a pointer option, I don't think ccur_fpu_to_H is that helpful as a name here, and cur_fpu_owner argument is misleading; don't we already have a function that does this?

Reply:
then again, this is probably copied from `ccur_vcpu_to_H` so I don't know anymore

Reply:
I'm somewhat surprised we don't have this function yet somewhere, it does seem generally useful for all C TCB pointers. +1 for giving it a more general name and argument name.

Reply:
Yeah, this was just following `ccur_vcpu_to_H`, which is a bit more complicated being about VCPUs and needing to take into account whether the VCPU is active or not. If I make it generic should it be moved somewhere else (and if so, any suggestions?)